### PR TITLE
refactor(scheduler): move [Signal] to [Stdune]

### PR DESCRIPTION
**Pull request by rgrinberg**
The `Signal` module is general purpose enough to live in `Stdune`. I've also expanded it to cover all known signals.

### CODE REVIEW

File: otherlibs/stdune/signal.ml [hunk 1]
```diff
@@ -1,37 +1,134 @@
...
+    ; (Sys, "SYS")
+    ; (Trap, "TRAP")
+    ; (Urg, "URG")
+    ; (Xcpu, "XCPU")
+    ; (Xfsz, "XFSZ")
     ]
   in
   fun n ->
     match List.assoc table n with
-    | None -> Printf.sprintf "%d\n" n
```
Comment (rgrinberg) [inline otherlibs/stdune/signal.ml:36]: This addition newline was probably just a bug.

File: otherlibs/stdune/signal.mli [hunk 2]
```diff
@@ -1,4 +1,43 @@
...
+  | Bus
+  | Poll
+  | Sys
+  | Trap
+  | Urg
+  | Xcpu
+  | Xfsz
+  | Unknown of int
+
 (** Convert a signal number to a name: "INT", "TERM", ... *)
```
Comment (snowleopard) [inline otherlibs/stdune/signal.mli:34]:
```suggestion
(** Get the signal's name, e.g. [Int] maps to "INT", [Term] to "TERM", etc. *)
```

File: otherlibs/stdune/signal.ml [hunk 3]
```diff
@@ -1,37 +1,134 @@
...
-    ; (sigvtalrm, "VTALRM")
-    ; (sigprof, "PROF")
-    ; (sigbus, "BUS")
-    ; (sigpoll, "POLL")
-    ; (sigsys, "SYS")
-    ; (sigtrap, "TRAP")
-    ; (sigurg, "URG")
-    ; (sigxcpu, "XCPU")
-    ; (sigxfsz, "XFSZ")
+    [ (Abrt, "ABRT")
```
Comment (snowleopard) [inline otherlibs/stdune/signal.ml:66]: Going via a table is weird. Why not use pattern matching?

Comment (rgrinberg) [reply otherlibs/stdune/signal.ml:66]: I don't remember to be honest. We usually have a table to be able to rerverse the map, but not here. I'll make it a pattern match.

File: otherlibs/stdune/signal.ml [hunk 4]
```diff
@@ -1,37 +1,125 @@
...
+  | Bus
+  | Poll
+  | Sys
+  | Trap
+  | Urg
+  | Xcpu
+  | Xfsz
+  | Unknown of int
+
+let all =
```
Comment (snowleopard) [inline otherlibs/stdune/signal.ml:32]: How about getting rid of this table too? We can do this as follows:

* Implement `of_int` using pattern matching.

* Implement `to_int` by creating an `Int.Map.t` and looking up in it.

Traversing lists on each call just doesn't seem right to me. (Although I may be overthinking this.)

Comment (rgrinberg) [reply otherlibs/stdune/signal.ml:32]: This happens so rarely that the performance gained negligible, but I can make it a hash table lookup both ways if you want.  At least we won't need to duplicate much code if we add a new signal.

Comment (snowleopard) [reply otherlibs/stdune/signal.ml:32]: I don't think any duplication will be needed with the above idea?

But we can also keep the current implementation if you prefer -- just please add some comments saying that it's currently pretty slow.

Comment (rgrinberg) [reply otherlibs/stdune/signal.ml:32]: If we implement `of_int` using pattern matching and `to_int` with a map lookup, we will need to modify two places when we add a new signal:

* The list `(int * t) list`which enumerates all the known signal
* The pattern match which converts signals to integers.

If we start with a list and generate two tables from it, we only need to update this single list.

Comment (snowleopard) [reply otherlibs/stdune/signal.ml:32]: I thought we don't need the list of known signals but I guess I was wrong.

**Review by snowleopard — APPROVED** [review]
Looks good!

**Pull request merged**